### ISSUE TITLE
Users with permission to a specific device cannot access it

### ISSUE THREAD

**Issue by kj7ice** [Ylianst/MeshCentral#4157]
**Describe the bug**
I granted remote control permission to a user for a specific device. This user does not have permission to any device groups or any other devices. Also this user is not a member of any user groups.

The user sees a message saying there are no device groups when they log in, and the device this user should have permission for is not listed anywhere. If the user is provided a direct link to the node, the page is blank where the general, desktop, events, etc. tabs should be.

**To Reproduce**
1. Create a new user.
2. Grant remote control permission on a specific device.
3. New user logs in.
4. See message "no device groups..."

**Expected behavior**
User should see the device in the list and be able to access it.

**Server Software:**
 - Server OS: Ubuntu 20.04
 - Network: WAN mode with nginx reverse proxy.
 - Version: 1.0.38
 - Node: 16.15.1
 - Browser: Google Chrome

**Remote Device (please complete the following information):**
 - Device: Desktop
 - OS: Windows 10

**Additional context**
I have not needed to assign permission to a specific device before, so I am unsure if this worked in prior versions. I've seen similar bug reports in the past that were resolved with a new release. Please let me know how I can troubleshoot further to help identify the issue?

**Comment by kj7ice** [comment]
I tested making the user a member of a user group and then granting device/node permission to the user group, and received the same result.

**Comment by silversword411** [comment]
Can you provide a screenshot of the permissions you set?

**Comment by kj7ice** [comment]
These are the permissions I have set for that user.
![image](https://user-images.githubusercontent.com/85096517/175193274-d67af47a-28a7-4ca0-a4a6-edf725ea6dbf.png)

I can also see the user listed on the General tab of the device.
![image](https://user-images.githubusercontent.com/85096517/175193578-5bf80420-a924-4695-a0fa-8f83b43d4407.png)

This is the message the user sees when logged in.
![image](https://user-images.githubusercontent.com/85096517/175193697-33715a0b-bf6c-4ae3-878e-8a9c097ca686.png)

Finally, here is the blank screen when the user navigates directly to the URL for the device/node.
![image](https://user-images.githubusercontent.com/85096517/175193803-97a04691-ea9d-4031-a3bc-5bc73b5f87c7.png)

**Comment by kj7ice** [comment]
I don't see any errors in the Chrome developer console. However, I do see these websocket messages sent from the server with empty lists. Not sure if that is useful at all?
```json
{"action":"meshes","meshes":[]}
{"action":"nodes","nodes":{}}

**Comment by silversword411** [comment]
I've seen that blank, with 3 buttons on the left once before.

Restart your server, use incognito...still the same result?

**Comment by kj7ice** [comment]
Yes, same result.

I forgot to mention that I am using MariaDB for the database in case that makes a difference.

**Comment by silversword411** [comment]
Use
https://ylianst.github.io/MeshCentral/meshcentral/debugging/
And check your logs for errors.


Find anything relevant?

**Comment by silversword411** [comment]
If you want me to login to your server and look at things, look me up on discord.

https://discord.gg/xcJPmCxYFA

**Comment by Ylianst** [comment]
I can't seem to make the problem happen. On the admin side:

![image](https://user-images.githubusercontent.com/1319013/175434267-9d97ba2a-c9cb-4f06-810e-1476932c3ab2.png)

On the user side:

![image](https://user-images.githubusercontent.com/1319013/175434321-7a08c098-41fd-46d2-9b35-731a29d128de.png)

not sure what to do at this point.

Having a white screen would indicate some sort of web page error on the user's side. There are really no browser console messages on the browser with the blank MeshCentral page? If there was something, that would really help.

**Comment by kj7ice** [comment]
Finally I took the time to clone my production server into a development environment and dive into this bug deeper. In summary, I found an issue with how the relevant MariaDB/MySQL query is constructed which leads to a syntax error. Understandably others could not replicate the issue since the bug is specific to the type of database in use. Perhaps this will apply to PostgreSQL also?

Starting in meshuser.js on [line 697](https://github.com/Ylianst/MeshCentral/blob/339e3efbefac28e2a24d88a00798e24fed95b223/meshuser.js#L697), I found that `docs` is undefined when the callback function is executed. That lead me to [line 1376](https://github.com/Ylianst/MeshCentral/blob/339e3efbefac28e2a24d88a00798e24fed95b223/db.js#L1376) of db.js where `err` is not null, and thus the if statement (which returns the nodes that the user has permission to) is not executed. Here is the output when I print `err` to the console (truncated node ID):
```
Error: (conn=1725, no: 1064, SQLState: 42000) You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') OR id IN ('node//...Ga$vkwo$KMsx40w@Gmp9AF...' at line 1
sql: SELECT doc FROM main WHERE type = ? AND domain = ? AND (extra IN (?) OR id IN (?)) - parameters:['node','',,node//...Ga$vkwo$KMsx40w@Gmp9AF...]
```
The query parameters look correct in that the user only has permission to one node and no meshes. However, the empty list `meshes = []` was not evaluating to an empty string, resulting in this SQL statement with an empty `IN ()` condition:
```sql
SELECT doc FROM main WHERE type = 'node' AND domain = '' AND (extra IN () OR id IN ('node//...'))
```
I'll submit a pull request that patches this bug by checking for an empty `meshes` list and setting it as an empty string. Open to feedback and maybe there is a more elegant solution!
